src/main/java/org/opensearch/knn/jni/FaissService.java

```diff
@@ -50,32 +50,64 @@ class FaissService {
     }
 
     /**
-     * Create an index for the native library The memory occupied by the vectorsAddress will be freed up during the
-     * function call. So Java layer doesn't need to free up the memory. This is not an ideal behavior because Java layer
-     * created the memory address and that should only free up the memory. We are tracking the proper fix for this on this
-     * <a href="https://github.com/opensearch-project/k-NN/issues/1600">issue</a>
+     * Initialize an index for the native library. Takes in numDocs to
+     * allocate the correct amount of memory.
      *
-     * @param ids array of ids mapping to the data passed in
-     * @param vectorsAddress address of native memory where vectors are stored
+     * @param numDocs number of documents to be added
```

Comment:
Does this need to be fixed? What if a user doesnt know up front?

Reply:
If the user doesn't know how many documents will be in the index, they could set numDocs to zero. All numDocs is used for is to make sure there is enough capacity to hold all of the data we will insert. I will add a comment here to make this clear 👍